crates/dbsp/src/storage/backend/posixio_impl.rs

```diff
@@ -210,9 +244,18 @@ impl Storage for PosixBackend {
         let mut files = self.files.borrow_mut();
 
         let mut fm = files.remove(&fd.0).unwrap();
-        let path = fm.path.clone();
         fm.flush()?;
         fm.file.sync_all()?;
+
+        // Remove the .mut extension from the file.
+        // I *believe* that on linux/(and less so in general POSIX) open fds are
+        // unaffected by a rename and remain valid. If I'm wrong we just have to close
+        // and open again after the rename.
```

Comment:
That's right, rename (and delete) don't affect open fds on Linux. POSIX says that a system can return EBUSY if "The file named by the path argument cannot be unlinked because it is being used by the system or another process and the implementation considers this an error." but that is an unusual implementation.